src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx

```diff
@@ -101,10 +102,10 @@ export class FieldsBreakdownScene extends SceneObjectBase<FieldsBreakdownSceneSt
     this.setState({
       fields: [
         { label: 'All', value: ALL_VARIABLE_VALUE },
-        ...(logsScene.state.detectedFields?.map((f) => ({
+        ...filterUsedLabelNames(logsScene, logsScene.state.detectedFields || []).map((f) => ({
```

Comment:
We could make `filterUsedLabelNames` accept `[]string | undefined` and then we would be able to remove all of `|| []` and `?? []`

Reply:
We could, but then the function signature would be wrong, because why would you call it with an undefined parameter. I would consider it a bad practice in the function definition.

Reply:
I understand the purpose, but the correct sematic approach, IMHO, should be to not call something when it should not be called, rather than have the function do nothing when nothing is passed.